presignedurl_tests.php

```diff
@@ -35,18 +35,26 @@
 
 echo $output->header();
 echo $output->heading(get_string('presignedurl_testing:page', 'tool_objectfs'));
+$delete = optional_param('delete', 0, PARAM_INT);
+$settingslink = \html_writer::link(new \moodle_url('/admin/settings.php?section=tool_objectfs'),
+    get_string('presignedurl_testing:objectfssettings', 'tool_objectfs'));
 
 $config = manager::get_objectfs_config();
 $support = false;
 if (!empty($config->filesystem)) {
     $fs = new $config->filesystem();
     $support = $fs->supports_presigned_urls();
 }
-$settingslink = \html_writer::link(new \moodle_url('/admin/settings.php?section=tool_objectfs'),
-    get_string('presignedurl_testing:objectfssettings', 'tool_objectfs'));
-
 if ($support) {
-
+    if ($delete === 1) {
+        $output->delete_presignedurl_tests_files();
+        echo $output->notification('Files deteled successfully', 'success');
+        $url = new \moodle_url('/admin/tool/objectfs/presignedurl_tests.php');
+        echo $output->heading(html_writer::link($url, 'Reload page'), 6);
+        echo $output->footer();
```

Comment:
If you render the footer you should probably just exit after that.

But in this case why not just move the logic further up so you don't need a reload after the delete?